### PR TITLE
Resolves kaazing/gateway#167: slow start ups on machines with low entropy

### PR DESCRIPTION
**Pull request by dpwspoon**
solved by having java set -Djava.security.egd=file:/dev/urandom on startup

### CODE REVIEW

File: distribution/src/main/gateway/bin/gateway.start [hunk 1]
```diff
@@ -41,9 +41,17 @@ GW_HOME=`cd "$PRGDIR/.." ; pwd`
...
 # of the GATEWAY_OPTS environment variable before calling this script.
-# The script itself should not be changed. For example, the setting
-# below sets the Java maximum memory to 512MB.
-[ -z "$GATEWAY_OPTS" ] && GATEWAY_OPTS="-Xmx512m"
+# The script itself should not be changed.
+if [ -z "$GATEWAY_OPTS" ]; then
+  # Sets the Java maximum memory to 512MB
+  GATEWAY_OPTS="-Xmx512m"
+  # Sets the java.security.edg to use /dev/urandom. By default /dev/random
+  # is used.  /dev/random is a blocking call which can hang on machines that
```
Comment (brennangaunce) [inline distribution/src/main/gateway/bin/gateway.start]: Suggest:

By default, Java uses /dev/random to gather entropy data for cryptographic needs.  However, using /dev/random can cause delays during Gateway startup, especially in virtualized environments.  /dev/urandom does not require collection of entropy data in subsequent runs.  See:  https://github.com/kaazing/gateway/issues/167

**Comment by jitsni** [comment]
Should we change to /dev/urandom file only docker environments ?

**Comment by dpwspoon** [comment]
We did consider this and thought it would be better to have in the generic gateway.start because this could hypothetically happen on VMs, cloud environments, and linux containers in general.  

Additionally, from what I can tell urandom is secure (reference [post1](http://security.stackexchange.com/questions/3936/is-a-rand-from-dev-urandom-secure-for-a-login-key) and [post2](http://www.2uo.de/myths-about-urandom/)).  

And then we should probably add documentation to deploying in production environment that if your server generates enough entropy remove this setting. PING @vjwang @robinzimmermann @sbadugu  (not sure how to properly tag that this is required, should I open a new issue?, I believe Robin and Satish were talking about this today)

**Comment by brennangaunce** [comment]
Yes - file a new issue to doc this appropriately.

**Pull request merged**